index.md

```diff
@@ -2,9 +2,286 @@
+This could be expanded to be a more permanent local storage solution, although there are limits as to how big a SQLite registry can usefully be.
+
+This approach does not work with the concept of allowing scientists to collaborate on their research since the datasets and registry would be completely private.
+It also does not work well with the concept of user batch {cite:p}`DMTN-223` where data are processed at the USDF but there is no way for a cloud RSP user to see the outputs if they are using a private Butler in the cloud.
+
+#### Butlers for Everyone
+
+An option proposed at the Joint Technical Meeting in February was to give everyone a personal butler PostgreSQL registry along with a personal object store bucket.[^oprah]
+This is intriguing because it allows for everyone to be completely independent, allows us to allocate users to specific servers, and would not require us to replicate all registry entries to every server.
+It would also be possible to create group-based butlers for people to collaborate on – if a scientist wants to make a dataset available to the group they can transfer the dataset to that butler using `Butler.transfer_from` in the normal way without changing the UUID of the dataset.
+
+Seeding these personal butlers would involve using `Butler.transfer_from` from the primary butler (but not copying any of the files, instead somehow enforcing a `direct` transfer mode be used), which leads to the possibility of someone seeding their butler with all of the datasets from the primary one.
+At that point you might have 10,000 or more personal replicas of the entire data release registry, which is less than ideal.
+It might be necessary to explicitly block the transfer of datasets from the primary butler to the user butler using `transfer_from` and only allow new datasets to be stored – a `butler.get` from the primary followed by a `butler.put` to the user butler can not be prevented, but this would result in the copied dataset using the user's quota and so would be self-limiting.
+If a user wants to query multiple collections across both butlers they would have to do that as independent queries.
+
+A more general problem with separate registries is that Butler can not currently query two registries simultaneously.
+To build a quantum graph involving some datasets from a data release and some from a user registry is not yet possible and would require considerable thought.
+We also need to understand if this is handled as two separate butlers from the user perspective or if the user has a single butler instance that is combined internally (either through the server being told to use multiple registries or the remote butler client talking to two distinct butlers internally).
+
+#### A single user registry
+
+A read-only registry has many advantages over a registry that is constantly receiving updates.
+In particular deployments of new replicas or restoring from backups or disaster recovery is very easy.
```

Comment:
The replication issue is the big one, but the other main advantage of separating the two is that we are in control of the performance characteristics of the large DB -- we know the size of all the tables, and we can test up-front against a fixed quantity of data to understand how we will perform under load.  If users can do whatever they want to it, all bets are off.

There's also the issue that the lifecycle of the two things is not the same -- the data release is going to be taken offline after two years, but we probably don't want to delete user data that quickly.